soc/arm/nordic_nrf/validate_base_addresses.c

```diff
@@ -133,57 +133,57 @@ CHECK_ADDRESS(DT_INST_0_NORDIC_NRF_REGULATORS_BASE_ADDRESS, NRF_REGULATORS);
 CHECK_ADDRESS(DT_INST_0_NORDIC_NRF_RNG_BASE_ADDRESS, NRF_RNG);
 #endif
 
-#if defined(DT_NORDIC_NRF_RTC_RTC_0_BASE_ADDRESS)
-CHECK_ADDRESS(DT_NORDIC_NRF_RTC_RTC_0_BASE_ADDRESS, NRF_RTC0);
+#if defined(DT_ALIAS_RTC_0_BASE_ADDRESS)
+CHECK_ADDRESS(DT_ALIAS_RTC_0_BASE_ADDRESS, NRF_RTC0);
 #endif
-#if defined(DT_NORDIC_NRF_RTC_RTC_1_BASE_ADDRESS)
-CHECK_ADDRESS(DT_NORDIC_NRF_RTC_RTC_1_BASE_ADDRESS, NRF_RTC1);
+#if defined(DT_ALIAS_RTC_1_BASE_ADDRESS)
+CHECK_ADDRESS(DT_ALIAS_RTC_1_BASE_ADDRESS, NRF_RTC1);
 #endif
-#if defined(DT_NORDIC_NRF_RTC_RTC_2_BASE_ADDRESS)
-CHECK_ADDRESS(DT_NORDIC_NRF_RTC_RTC_2_BASE_ADDRESS, NRF_RTC2);
+#if defined(DT_ALIAS_RTC_2_BASE_ADDRESS)
+CHECK_ADDRESS(DT_ALIAS_RTC_2_BASE_ADDRESS, NRF_RTC2);
 #endif
 
-#if defined(DT_NORDIC_NRF_SAADC_ADC_0_BASE_ADDRESS)
-CHECK_ADDRESS(DT_NORDIC_NRF_SAADC_ADC_0_BASE_ADDRESS, NRF_SAADC);
+#if defined(DT_ALIAS_ADC_0_BASE_ADDRESS)
+CHECK_ADDRESS(DT_ALIAS_ADC_0_BASE_ADDRESS, NRF_SAADC);
 #endif
 
-#if defined(DT_NORDIC_NRF_SPI_SPI_0_BASE_ADDRESS)
-CHECK_ADDRESS(DT_NORDIC_NRF_SPI_SPI_0_BASE_ADDRESS, NRF_SPI0);
+#if defined(DT_ALIAS_SPI_0_BASE_ADDRESS)
```

Comment:
This won't work, and CI fails because of this, as previously the generated macro contained in its name the type of the peripheral to be checked (SPI, SPIM, or SPIS) and presence of the macro implied that the corresponding `NRF_*` symbol would be defined. Now regardless of the peripheral type, a common macro name is used (`DT_ALIAS_SPI_0_BASE_ADDRESS` in this case), thus these checks need to be updated. I think the easiest way will be to just add `&& defined(NRF_SPI0)` here and corresponding conditions below (for SPIMx, SPISx, TWIx, TWIMx, TWISx, UART0, and UARTEx).